File: core/src/main/java/com/truenine/component/core/dev/BetaTest.java
package com.truenine.component.core.dev;

import java.lang.annotation.*;

/**
 * 该注解标志这这个类、方法、参数……
 * <br/>
 * 可能在未来版本移除或者变更，因为底层依赖不稳定
 * <br/>
 * 或者目前也没有找到好的解决方式
 *
 * @author TrueNine
 * @date 2022-10-26
 */
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target({
  ElementType.ANNOTATION_TYPE,
  ElementType.CONSTRUCTOR,
  ElementType.FIELD,
  ElementType.METHOD,
  ElementType.TYPE})
@Documented
public @interface BetaTest {
}

File: core/src/main/kotlin/com/truenine/component/core/annotations/BetaTest.kt
package com.truenine.component.core.annotations

import java.lang.annotation.Inherited

/**
 * 该注解标志这这个类、方法、参数……
 * <br></br>
 * 可能在未来版本移除或者变更，因为底层依赖不稳定
 * <br></br>
 * 或者目前也没有找到好的解决方式
 *
 * @author TrueNine
 * @since 2022-10-26
 */
@Inherited
@Retention(AnnotationRetention.RUNTIME)
@Target(
  AnnotationTarget.ANNOTATION_CLASS,
  AnnotationTarget.CONSTRUCTOR,
  AnnotationTarget.FIELD,
  AnnotationTarget.FUNCTION,
  AnnotationTarget.PROPERTY_GETTER,
  AnnotationTarget.PROPERTY_SETTER,
  AnnotationTarget.CLASS
)
@MustBeDocumented
annotation class BetaTest
